packages/common/src/content/manageSchedule.ts

```diff
@@ -44,52 +40,102 @@ export const getSchedule = async (
 
 /**
  * Set the schedule for an item
- * @param item The item to set the schedule for
+ * @param itemId The item to set the schedule for
  * @param schedule The schedule to set
  * @param requestOptions The request options needed to get the HubApiUrl
  */
 export const setSchedule = async (
-  item: IItem,
+  itemId: string,
   schedule: IHubSchedule,
   requestOptions: IRequestOptions
-): Promise<void> => {
-  if (schedule.mode) {
-    delete schedule.mode;
-  }
-  const url = `${getHubApiUrl(requestOptions)}/api/download/v1/items/${
-    item.id
-  }/schedule`;
+): Promise<any> => {
+  const body = cloneObject(schedule);
+  delete body.mode;
+  const url = schedulerApiUrl(itemId, requestOptions);
   const options = {
     method: "POST",
     headers: {
       "Content-Type": "application/json",
       accept: "application/json",
     },
     body: JSON.stringify({
-      ...schedule,
-      itemId: item.id,
+      ...body,
+      itemId,
     }),
   };
-  await fetch(url, options);
+  const response = await fetch(url, options);
+  return await response.json();
 };
 
 /**
  * Delete the schedule for an item
- * @param item The item to delete the schedule for
+ * @param itemId The item to delete the schedule for
  * @param requestOptions The request options needed to get the HubApiUrl
  */
 export const deleteSchedule = async (
-  item: IItem,
+  itemId: string,
   requestOptions: IRequestOptions
-): Promise<void> => {
-  const url = `${getHubApiUrl(requestOptions)}/api/download/v1/items/${
-    item.id
-  }/schedule`;
+): Promise<any> => {
+  const url = schedulerApiUrl(itemId, requestOptions);
   const options = {
     method: "DELETE",
     headers: {
       accept: "application/json",
     },
   };
-  await fetch(url, options);
+  const response = await fetch(url, options);
+  return await response.json();
+};
+
+/**
+ * Checks if the content schedule should be updated and updates it if necessary
+ * @param content The content to check and update the schedule for (should include any new schedule information)
+ * @param requestOptions The request options needed to get the HubApiUrl
+ */
+export const maybeUpdateSchedule = async (
+  content: IHubEditableContent,
+  requestOptions: IRequestOptions
+) => {
+  const currentSchedule = await getSchedule(content.id, requestOptions);
+
+  // if no schedule is set and incoming schedule is automatic, do nothing
+  if (content.schedule.mode === "automatic" && currentSchedule === null) {
+    return false;
```

Comment:
Whenever we are working in typescript, we want to make sure that we are very deliberate with our return types. I notice that we don't have an explicit return type on this function and the `deleteSchedule`, `setSchedule`, etc. have a return type of `any`.

Try and come up with a consistent pattern (e.g., always returning a boolean, returning an `IMaybeUpdateScheduleResponse`, etc.) and let's talk when you do.